### PR TITLE
(#565) Update prow linter to validate env vars

### PR DESCRIPTION
**Pull request by gwesterfieldjr**
#565 https://github.com/aws/eks-anywhere/issues/565

* Added new EnvVar check to linter
* Moved hard coded JobConstants to config/job-constants.yaml

@abhay-krishna @vignesh-goutham  @jaxesn @ndeksa @danbudris 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by gwesterfieldjr** [comment]
The only env var listed in the task to check was ARTIFACTS_BUCKET. To add others in the future we simple need to add it to the env list in the job-constants.yaml

File: scripts/lint_prowjobs/main.go [hunk 1]
```diff
@@ -145,23 +173,37 @@ func getFilesChanged(gitRoot string, pullBaseSha string, pullPullSha string) ([]
...
 	fileContents, fileReadError := ioutil.ReadFile(filePath)
-	unmarshaledJobConfig.FileContents = string(fileContents)
 
-	fileUnmarshalError := yaml.Unmarshal(fileContents, &jobConfig)
-	unmarshaledJobConfig.ProwjobConfig = jobConfig
+	if fileReadError != nil {
+		log.Fatalf("Error reading contents of %s: %v", filePath, fileReadError)
+	}
+
+	unmarshalError := yaml.Unmarshal([]byte(fileContents), data)
```
Comment (abhay-krishna) [inline scripts/lint_prowjobs/main.go:197]: nit: The `ioutil.ReadFile` method returns a byte slice, so is it necessary to cast it to type byte slice?

Comment (gwesterfieldjr) [reply scripts/lint_prowjobs/main.go]: Fixed.

**Comment by abhay-krishna** [comment]
> The only env var listed in the task to check was ARTIFACTS_BUCKET. To add others in the future we simply need to add it to the env list in the job-constants.yaml

Can we also add support for the env vars [here](https://github.com/aws/eks-anywhere-prow-jobs/blob/main/jobs/aws/eks-anywhere/eks-anywhere-e2e-presubmits.yaml)?

**Comment by gwesterfieldjr** [comment]
@abhay-krishna yes! Thanks for the additional vars!

**Comment by abhay-krishna** [comment]
Some more [here](https://github.com/aws/eks-anywhere-prow-jobs/blob/main/jobs/aws/eks-anywhere/eks-anywhere-e2e-cleanup-periodics.yaml) with some overlap.

**Comment by gwesterfieldjr** [comment]
@abhay-krishna So the values for TEST_ROLE_ARN env var in eks-anywhere/eks-anywhere-e2e-presubmits.yaml and eks-anywhere/eks-anywhere-e2e-cleanup-periodics.yaml are different. I'm working under the assumption that we are trying to enforce a consistent value for all the env vars listed in job-contants.yaml. With that said, which TEST_ROLE_ARN value is the correct one for us to enforce? Or If they should be different values which env var do you want to rename? And if we rename it I assume there are some down stream jobs that we would have to update to reflect the new name as well. Please let me know your thoughts.

**Comment by gwesterfieldjr** [comment]
> Some more [here](https://github.com/aws/eks-anywhere-prow-jobs/blob/main/jobs/aws/eks-anywhere/eks-anywhere-e2e-cleanup-periodics.yaml) with some overlap.

I pushed the additional vars, but one of them has an inconsistent value in 2 different job files. Please see my other comment for more details so we can take appropriate action.

**Review by abhay-krishna — COMMENTED** [review]
Thanks for this @gwesterfieldjr 🎉 
/lgtm
/approve

**Pull request merged**